samples/nrf9160/serial_lte_modem/src/slm_at_host.c

```diff
@@ -276,11 +277,13 @@ static void cmd_send(struct k_work *work)
 	}
 
 done:
+	k_sleep(100); /* allow time for TX DMA */
```

Comment:
in order to get rid of this delay you need to do following:
- create another semaphore which will ensure that calling rx enabled occurs after disabling is completed.
- in uart event handler add handing of `UART_RX_DISABLED` event. In that event give that semaphore
- here replace k_sleep with sem_take.

Reasoning: Uart disabling is asynchronous. When `uart_rx_disable` is called it is not yet completed. There will be couple of event coming from uart (e.g. releasing buffers) and finally `UART_RX_DISABLED` which indicates that rx is off. Only after that event, uart can be reenabled again.